src/garage/experiment/local_runner.py

```diff
@@ -274,26 +298,29 @@ def step_epochs(self):
                 runner.step_itr += 1
 
         """
-        try:
-            self._start_worker()
-            self._start_time = time.time()
-            self.step_itr = (self.train_args.start_epoch *
-                             self.train_args.n_epoch_cycles)
-            self.step_path = None
-
-            for epoch in range(self.train_args.start_epoch,
-                               self.train_args.n_epochs):
-                self._itr_start_time = time.time()
-                with logger.prefix('epoch #%d | ' % epoch):
-                    yield epoch
-                    save_path = (self.step_path
-                                 if self.train_args.store_paths else None)
-                    self.save(epoch, save_path)
-                    self.log_diagnostics(self.train_args.pause_for_plot)
-                    logger.dump_all(self.step_itr)
-                    tabular.clear()
-        finally:
-            self._shutdown_worker()
+        self._start_worker()
+        self._start_time = time.time()
+        self.step_itr = self._stats.total_itr
+        self.total_env_steps = self._stats.total_env_steps
+        self.step_path = None
+
+        for epoch in range(self._train_args.start_epoch,
+                           self._train_args.n_epochs):
+            self._itr_start_time = time.time()
+            with logger.prefix('epoch #%d | ' % epoch):
+                yield epoch
+                save_path = (self.step_path
+                             if self._train_args.store_paths else None)
+
+                self._stats.last_path = save_path
+                self._stats.total_epoch = epoch
```

Comment:
i think that you could make this significantly simpler by defining a `Stats` or `Metrics` class.

For instance, you could define some quantities as counters and only support incrementing them. You could also log the entire class `tabular.log(self._stats)`.

Reply:
Let's make it in another PR. This will involve the problem if existing stats should be included in this new class.